### PR TITLE
Feature deploy

### LINKED ISSUE #27: Admin demo missing?
Where can we find admin demo? It seems to be missing here https://demo.api-platform.com/admin

TIA

### PR DESCRIPTION
**Pull request by vincentchalamon**
- [x] Fix trusted hosts issue
- [x] Add [external-dns](https://github.com/kubernetes-incubator/external-dns)
- [x] Handle dynamic domain name (such as `demo-feature-deploy.api-platform.com`)
- [x] Configure Cloudflare
- [x] Enable/configure feature deploy
- [x] Update documentation
- [x] Create/update DNS for buckets
- [x] Update docker db image to `bitnami/postgresql`
- [x] Remove [TODO](https://github.com/api-platform/demo/pull/95/files#diff-cba52990deff0a97e3abb262289a6ad7R5)
- [x] Remove useless namespaces/releases
- [x] "_Remove namespace/release master_": release/namespace "master" should not be removed

Fix #27 & #96
Doctrine does not longer support DB-generated UUIDs: https://github.com/doctrine/dbal/pull/3212. Rollback to `ramsey/uuid-doctrine`

### CODE REVIEW

File: api/helm/api/values.yaml [hunk 1]
```diff
@@ -99,28 +98,49 @@ mercure:
...
+    email: ""
+  rbac:
+    create: true
+
 ingress:
   annotations:
-    kubernetes.io/ingress.global-static-ip-name: demo-ip
-    # kubernetes.io/ingress.class: gce
+    # kubernetes.io/ingress.global-static-ip-name: chart-ip
+    kubernetes.io/ingress.class: gce
```
Comment (dunglas) [inline api/helm/api/values.yaml]: can't this be removed as it's the default on GCE?

File: api/helm/api/values.yaml [hunk 2]
```diff
@@ -99,28 +98,49 @@ mercure:
...
     # kubernetes.io/tls-acme: "true"
   tls:
     # Secrets must be manually created in the namespace.
```
Comment (dunglas) [inline api/helm/api/values.yaml]:
```suggestion
    # Secrets must be manually created in the namespace, you can also use cert-manager.
```

**Comment by vincentchalamon** [comment]
Feature deploy is working fine, please check the following domains:
https://demo-feature-deploy.api-platform.com
https://demo-mercure-feature-deploy.api-platform.com
https://demo-admin-feature-deploy.api-platform.com
https://demo-client-feature-deploy.api-platform.com

File: README.md [hunk 3]
```diff
@@ -16,3 +16,19 @@ Loading Fixtures
...
+Everything is pre-configured. Edit the `ci/.env` file to set your project parameters, and declare the following secured
+environment variables in your CI:
+
+ * `PROJECT_ID`: GCP project id (i.e: `api-platform-demo-123456`)
+ * `CI_SERVICE_ACCOUNT`: GCP service account
+ * `CI_SERVICE_ACCOUNT_KEY`: GCP service account key
+ * `CF_API_KEY`: Cloudflare API key
+ * `CF_API_EMAIL`: Cloudflare email
+
+**Important: do not check "_Display value in build log_"!**
```
Comment (Mathieudewet) [inline README.md]: **Important: do not check "_Display value in build log_" for security reason !**

File: ci/.env [hunk 4]
```diff
@@ -3,8 +3,29 @@
...
 # Both CI will make tests but only the one specified will deploy.
 # Current available choices are travis and circleci.
-export CURRENT_CI='travis'
-export MULTI_BRANCH=0
-export REPOSITORY='api-platform/demo'
-export DEPLOYMENT_BRANCH='master'
-export PROD_DNS='demo.api-platform.com'
+export CURRENT_CI=travis
+
+# If you want to deploy feature branches, set FEATURE_DEPLOY value to 1.
```
Comment (Mathieudewet) [inline ci/.env]: It's already the case, should we say "If you doesn't want to deploy feature branches, set FEATURE_DEPLOY value to 0." instead ?

File: ci/deploy [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    --set postgresql.postgresqlPassword=$DATABASE_PASSWORD \
+    --set ingress.hosts.api.host=$API_ENTRYPOINT \
+    --set ingress.hosts.mercure.host=$MERCURE_ENTRYPOINT \
+    --set mercure.subscribeUrl="https://${MERCURE_ENTRYPOINT}/hub" \
+    --set external-dns.cloudflare.apiKey=$CF_API_KEY \
+    --set external-dns.cloudflare.email=$CF_API_EMAIL \
+    --set corsAllowOrigin=$CORS_ALLOW_ORIGIN \
+    --set mercure.corsAllowOrigin=$CORS_ALLOW_ORIGIN \
+    --set trustedHosts=$TRUSTED_HOSTS \
+    --set mercure.domainFilters="{$DOMAIN}"
```
Comment (Mathieudewet) [inline ci/deploy]: --set mercure.domainFilters="${DOMAIN}"

Comment (Mathieudewet) [reply ci/deploy]: Or even `$DOMAIN`

Comment (vincentchalamon) [reply ci/deploy]: DOMAIN must be an array

**Pull request merged**